markdown_it/tree.py

```diff
@@ -211,27 +211,22 @@ def _set_children_from_tokens(self, tokens: Sequence[Token]) -> None:
         while reversed_tokens:
             token = reversed_tokens.pop()
 
-            if token.nesting == 0:
-                child = self._make_child(token=token)
-                if token.children:
-                    child._set_children_from_tokens(token.children)
+            if not token.nesting:
+                self._add_child([token])
                 continue
-
-            assert token.nesting == 1
+            if token.nesting != 1:
+                raise ValueError("Invalid token nesting")
```

Comment:
it would be nice here to identify where in the token stream the issue is.
Maybe at the start of this function set `length = len(tokens)`, then here do `index = length - len(reversed_tokens)` and add this to the exception message

Reply:
Note that due to the nested nature of the tree, there's a very good chance `tokens` won't be the same stream that the programmer input to `SyntaxTreeNode` init but rather a nested sub-stream. In this case showing the index could potentially be more confusing than helpful. I'm also a bit hesitant to add logic just for the sake of an exception message that, in software working expectedly, should never be raised.

Can add this if you still think its a good idea.

Reply:
ah yeh fair I think, if only for myst-parser, I want to have a new exception (i.e. that inherits `ValueError`) that stores the token, so I can report maybe the line number (from token.map) or whole token, but I can add that in a separate PR